### DISCUSSION TITLE
Couldn't create omegaconf with AWS tokens

### DISCUSSION THREAD

**Discussion by igovindarajan-fincad** [omry/omegaconf#811] [General]
AWS tokens is of the form `${Token[TOKEN.186]}`. Latest omegaconf complaints as below.
`omegaconf.errors.GrammarParseError: missing BRACKET_CLOSE at '.'` when I try the following code:

```
>>> import omegaconf
>>> omegaconf.OmegaConf.create({"s3":"${Token[TOKEN.186]}"}) 
```

This was working fine until `2.1.0.dev19`. Can this be supported in the next version. By which omegaconf latest versions can be used in AWS CDK projects/repo. Thank you!

**Comment by Jasha10** [comment]
Try escaping the dollar sign with a backslash:
```python
>>> omegaconf.OmegaConf.create({"s3":"\${Token[TOKEN.186]}"})
```

For in-depth information on the omegaconf interpolation grammar, including what characters to escape and when, you can look at:
- there is a great PR #792 that will soon be merged, which updates the omegaconf docs with more details on the grammar
- You can look at the grammar files [omegaconf/grammar/OmegaConfGrammarLexer.g4](https://github.com/omry/omegaconf/blob/master/omegaconf/grammar/OmegaConfGrammarLexer.g4) and [omegaconf/grammar/OmegaConfGrammarParser.g4](https://github.com/omry/omegaconf/blob/master/omegaconf/grammar/OmegaConfGrammarParser.g4)

**Reply by igovindarajan-fincad** [reply]
This includes escaped backslash while retrieving the value. Is this intended?


```
>>> import omegaconf
>>> conf = omegaconf.OmegaConf.create({"s3":"\${Token[TOKEN.186]}"})
>>> print(conf.s3)
\${Token[TOKEN.186]}
```

**Reply by omry** [reply]
Check your OmegaConf version.

**Reply by Jasha10** [reply]
FYR @igovindarajan-fincad here is the behavior with OmegaConf 2.1.1
```python
>>> conf = omegaconf.OmegaConf.create({"s3":"\${Token[TOKEN.186]}"})
>>> print(conf.s3)
${Token[TOKEN.186]}
>>> omegaconf.__version__
'2.1.1'
```

**Reply by igovindarajan-fincad** [reply]
My bad, it works. Thanks.
